relentless/optimize/method.py

```diff
@@ -177,41 +203,80 @@ def __init__(self, abs_tol, max_iter):
         self.abs_tol = abs_tol
         self.max_iter = max_iter
 
-    def find(self, objective, start, end):
+    def find(self, objective, start, end, scale=1.0):
         """Apply the line search algorithm to take the optimal step.
 
         Note that the objective function is kept at its initial state, and the
         function evaluted after taking the optimal step is returned separately.
 
         Parameters
         ----------
-        objective : :class:`~relentless.optimize.ObjectiveFunction`
+        objective : :class:`~relentless.optimize.objective.ObjectiveFunction`
             The objective function for which the line search is applied.
-        start : :class:`~relentless.optimize.ObjectiveFunctionResult`
+        start : :class:`~relentless.optimize.objective.ObjectiveFunctionResult`
             The objective function evaluated at the start of the search interval.
-        end : :class:`~relentless.optimize.ObjectiveFunctionResult`
+        end : :class:`~relentless.optimize.objective.ObjectiveFunctionResult`
             The objective function evaluated at the end of the search interval.
+        scale : float or dict
+            A scalar scaling parameter or scaling parameters (:math:`\mathbf{X}`)
+            keyed on one or more `~relentless.optimize.objective.ObjectiveFunction`
+            design variables (defaults to ``1.0``, so that the variables are unscaled).
 
         Raises
         ------
+        ValueError
+            If the scaling parameters are not positive.
+        ValueError
+            If the start and the end of the search interval are identical.
+        KeyError
+            If the scaling parameters are defined on a variable which is not in
+            the objective function.
         ValueError
             If the defined search interval is not a descent direction.
 
         Returns
         -------
-        :class:`~relentless.optimize.ObjectiveFunctionResult`
-            The objective function evaluated at the new, 'optimal' location.
+        :class:`~relentless.optimize.objective.ObjectiveFunctionResult`
+            The objective function evaluated at the new, "optimal" location.
 
         """
+        # check if valid scaling
```

Comment:
I would suggest that you coerce the scaling parameters into a `KeyedArray` here so that you can use this math later.